moved pdf file to separate folder

kostyfisik opened the pull request:
Чтобы не было конфликтов с обновляемыми pdf и с теми, которые предлагаются в качестве примера предлагаю переместить результат компиляции в отдельную папку. Если необходимо обновить примеры - их явным образом надо будет переписать поверх файлами из папок Dissertation и Synopsis

Lenchik commented:
Что-то смысла не пойму. Скомпилировались - вот и обновленный пример. Зачем что-то ещё перемещать?

kostyfisik commented:
Оно конфликтует с написанием дисера в соседней ветке. Либо каждый раз, когда пишешь дисер, надо комитить pdf, либо откатывать их  в исходное, либо прятать, либо менять путь для файла дисера в своей ветке...

Сейчас развитие шаблона крайне прямолинейно, если возникнет несколько разрабатываемых веток (например под разные госты, или разные движки теха, например посмотреть возможности LuaTeX), то будет аналогичная проблема.

Lenchik commented:
Не будет веток в обозримой перспективе.
Возможности всех трёх компиляторов и так работают, чтобы не надо было ветки плодить.
Вопрос коммита пдф решается из гита. Наверняка есть команды, чтобы не всё изменённое коммитить. Почему так думаю? Вот SmartGit позволяет выбрать что коммитить, а он - продвинутый GUI над git.
И потом - был же уже PR, игнорирующий пдф.
Вообще непонятно, если несколько веток, то переход с ветки на ветку перетаскивает и состояние их. Так что ценный пдф диссера из соседней ветки и пдф из ветки шаблона вообще не должны влиять друг на друга. 
Вроде бы git для таких фокусов и задумывался

kostyfisik commented:
Git ругается, если есть незакомиченные изменения в отслеживаемых файлах. Собственно PR с игнором PDF решил проблему (и поднял вопрос про размер .git), до тех пор пока не поменялся pdf  в мастере. Предлагаемое изменение просто зафиксирует уже существующую роль pdf в шаблоне - в качестве примера. Если человек безо всякого создания веток возьмёт шаблон и начнёт его править - то его вариант пример затрёт. В случае каких либо непонятных ошибок (как у меня в #80 ) чтобы посмотреть оригинал и убедиться, что там этой ошибки не было - придётся откатывать файл (и тогда не получится их смотреть рядом, по крайне мере в линуксе штатный просмотровшик автоматом перерисовывает измененный файл) или качать пример с сайта заново.

В общем предлагаю добавить некой самодокументированности в код, когда примеры лежат в папке "Примеры", а не как побочный эффект компиляции по местам.

Lenchik commented:
Лишнее телодвижение для комиттеров.
Лучше уж делать свой чистовой билд диссера в отдельную папку каким-нибудь батником/makefile, а "потроха" пусть пока потрохами полежат. А потом и потроха можно было бы попричесать (#67).
Затирание оригинального пдф - беда всех умолчальных .tex примеров с интернета - качаешь результат.

kostyfisik commented:
Так оно и не плохо! Ибо ничего хорошего в частом комите бинарников нет, а так случайно бинарник не зашлёшь. Только в том случае, если он реально поменялся, чего не происходило уже с пол-года. Январский вариант https://github.com/AndreyAkinshin/Russian-Phd-LaTeX-Dissertation-Template/blob/04ac159f12b59a7133134218c2adf2b09812ec0f/Synopsis/synopsis.pdf по внешнему виду (при беглом просмотре "О! Это то что я искал для шаблона" ) мало отличается от сегодняшнего  https://github.com/AndreyAkinshin/Russian-Phd-LaTeX-Dissertation-Template/blob/master/Synopsis/synopsis.pdf

dustalov commented:
Не читал обсуждение, но надеюсь, что никто не собирается плодить отдельные ветки под разные компиляторы. Спасибо.

kostyfisik commented:
Откуда этот страх перед ветками? Git для того и был придуман, чтобы ветвиться и сливаться было просто. Тем не менее, ветвиться под разные компиляторы вроде никто и не собирается.

AndreyAkinshin commented:
LaTeX-шаблон — это не тот проект, в котором ветки нужны для чего-нибудь кроме PR. Обсуждали это уже несколько раз, пришли к выводу, что нужно держать единственную master-ветку, которая будет включать в себя все полезные наработки. Идея делать ветки под разные госты/движки не особо хороша, слишком уж много проблем она за собой тянет, лучше пытаться держать всё в мастере.

kostyfisik commented:
Хорошо, понял. Я правильно понимаю, что это только моя проблема, писать свой дисер в соседней ветке, отдавая все доработки в апстим? И как следствие накладывая все обновления из апстрима на свою мастер ветку и ветку с дисером?

AndreyAkinshin commented:
В таком сценарии я не вижу проблем, всё должно легко делаться.

kostyfisik commented:
Может я что-то не так делаю? Этот PR предлагается, чтобы избежать ошибки внизу лога. Из ветки upstream-master я утром делал PR #82 

tig@tig-laptop3:~/symlink/diser$  **git co diser-Ladutenko**
Already on 'diser-Ladutenko'
Your branch is ahead of 'origin/diser-Ladutenko' by 3 commits.
  (use "git push" to publish your local commits)
tig@tig-laptop3:~/symlink/diser$ **make**
make -C Synopsis

... Много вывода

Latexmk: All targets (dissertation.pdf) are up-to-date
make[1]: Leaving directory '/home/tig/KOsty/Documents/Study/kand_diser_ITMO/LaTeX/Russian-Phd-LaTeX-Dissertation-Template/Dissertation'
tig@tig-laptop3:~/symlink/diser$ **git diff**
diff --git a/Dissertation/dissertation.pdf b/Dissertation/dissertation.pdf
index b156ba3..edb925b 100644
Binary files a/Dissertation/dissertation.pdf and b/Dissertation/dissertation.pdf differ
diff --git a/Synopsis/synopsis.pdf b/Synopsis/synopsis.pdf
index f57b75d..db5ba63 100644
Binary files a/Synopsis/synopsis.pdf and b/Synopsis/synopsis.pdf differ
tig@tig-laptop3:~/symlink/diser$ **git co upstream-master**
error: Your local changes to the following files would be overwritten by checkout:
	Synopsis/synopsis.pdf
Please, commit your changes or stash them before you can switch branches.
Aborting

kostyfisik commented:
co - алиас для checkout

AndreyAkinshin commented:
Нужно просто сделать revert (или stash) файлов synopsis.pdf и dissertation.pdf (если их не предполагается включать в PR).

kostyfisik commented:
Оно в итоге оказывается не так просто. Если в master перетаскивание вывода что-то ломает - я просто перетащу оригинальные pdf  в ветке дисера, чтобы не мешались. Мою проблему эту решило, проверил на том же переключении.

tig@tig-laptop3:~/symlink/diser$ **git stash**
Saved working directory and index state WIP on diser-Ladutenko: 05332b0 Merge remote-tracking branch 'upstream/master' into diser-Ladutenko
HEAD is now at 05332b0 Merge remote-tracking branch 'upstream/master' into diser-Ladutenko
tig@tig-laptop3:~/symlink/diser$ **git co upstream-master** 
Switched to branch 'upstream-master'
Your branch is up-to-date with 'origin/upstream-master'.
tig@tig-laptop3:~/symlink/diser$ **make**
make -C Synopsis

... много вывода

Latexmk: All targets (dissertation.pdf) are up-to-date
make[1]: Leaving directory '/home/tig/KOsty/Documents/Study/kand_diser_ITMO/LaTeX/Russian-Phd-LaTeX-Dissertation-Template/Dissertation'
tig@tig-laptop3:~/symlink/diser$ **git co diser-Ladutenko** 
error: Your local changes to the following files would be overwritten by checkout:
	Synopsis/synopsis.pdf
Please, commit your changes or stash them before you can switch branches.
Aborting
tig@tig-laptop3:~/symlink/diser$ **git stash**
Saved working directory and index state WIP on upstream-master: 8cec114 add distclean target to Makefiles
HEAD is now at 8cec114 add distclean target to Makefiles
tig@tig-laptop3:~/symlink/diser$ git co diser-Ladutenko 
Switched to branch 'diser-Ladutenko'
Your branch is ahead of 'origin/diser-Ladutenko' by 3 commits.
  (use "git push" to publish your local commits)
tig@tig-laptop3:~/symlink/diser$ **git stash pop**
warning: Cannot merge binary files: Synopsis/synopsis.pdf (Updated upstream vs. Stashed changes)
Auto-merging Synopsis/synopsis.pdf
CONFLICT (content): Merge conflict in Synopsis/synopsis.pdf
